src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/BotSegment.cs

```diff
@@ -232,5 +234,170 @@ public void WriteToStringBuilder(StringBuilder stringBuilder)
+         * </summary>
+         */
+        public static List<BotSequenceEntry> LoadAllSegments()
+        {
+            var segments = new List<BotSequenceEntry>();
+            
+#if UNITY_EDITOR
+            const string sequencePath = "Assets/RegressionGames/Resources/BotSegments";
+            if (!Directory.Exists(sequencePath))
+            {
+                return new List<BotSequenceEntry>();
+            }
+
+            segments = LoadSegmentsInDirectory(sequencePath);
+#else
+            // 1. check the persistentDataPath for segments
+            var persistentDataPath = Application.persistentDataPath + "/BotSegments";
+            if (Directory.Exists(persistentDataPath))
+            {
+                segments = LoadSegmentsInDirectoryForRuntime(persistentDataPath);
+            }
+        
+            // 2. load Segments from Resources, and ensure that no Segments have duplicate names
+            const string runtimePath = "BotSegments";
+            segments = segments.Concat(LoadSegmentsInDirectoryForRuntime(runtimePath)).ToList();
+#endif
+
+            return segments;
+        }
+        
+        /**
+         * <summary>
+         * Recursively look through directories for Segement files, and load them
+         * </summary>
+         * <param name="path">
+         * Directory to search for Segments
+         * </param>
+         */
+        private static List<BotSequenceEntry> LoadSegmentsInDirectory(string path)
+        {
+            var results = new List<BotSequenceEntry>();
+            var directories = Directory.EnumerateDirectories(path);
+            foreach (var directory in directories)
+            {
+                var files = Directory.GetFiles(directory, "*.json");
+                foreach (var fileName in files)
+                {
+                    using var sr = new StreamReader(File.OpenRead(fileName));
+                    var result = (fileName, sr.ReadToEnd());
+                    var segment = ParseSegment(result.Item2, fileName);
+                    if (segment != null)
+                    {
+                        results.Add(segment);
+                    }   
+                }
+
+                // recurse into the next level of directories
+                results = results.Concat(
+                    LoadSegmentsInDirectory(directory)
+                ).ToList();
+            }
+
+            return results;
+        }
+        
+        /**
+         * <summary>
+         * Recursively look through directories for Segement files, and load them (in a build of the game)
+         * </summary>
+         * <param name="path">
+         * Directory to search for Segments
+         * </param>
+         */
+        private static List<BotSequenceEntry> LoadSegmentsInDirectoryForRuntime(string path)
+        {
+            var results = new List<BotSequenceEntry>();
+            var directories = Directory.EnumerateDirectories(path);
```

Comment:
Short Term
This API is not usable in the runtime environment for resources.  (or any File/Directory API)

I would expect you to be able to load these using only the resources api but only from that top level path.


Longer term:
I will create a task -> We'll need to make this a build time operation so we have more control and store the listing of our segments/sequences/etc into a scriptable object that we can then utilize here later instead of digging for resources / folders at runtime.

Reply:
To clarify, regarding the short term: Should I expect that the Bot Segments I want to load will only be at the top level path? 

When loading Segments in editor mode, the Segments are nested into different folders

Reply:
Short (from planning).. runtime builds just won't work well until I do the build time parsing task.  So i would just get the directory api / loop out of this PR so it doesn't crash/do weird stuff.

Reply:
I removed the use of the Directory function, so now we're just searching the `BotSegments` path only. It seems to load all of the Segments/SegmentLists in a build of RGBossRoom. So that's nice!